### ISSUE TITLE
Unable to compile after trying to update to 0.24.0: Unable to find method GenerateMultiplatformResourcesTask.onlyIf

### ISSUE THREAD

**Issue by bravo-delta** [icerockdev/moko-resources#732]
First of all: Thanks for all the efforts that went into the 0.24.0 Update

I tried to update my project to 0.24.0 but during compilation the following error is thrown:
`java.lang.NoSuchMethodError: 'void dev.icerock.gradle.tasks.GenerateMultiplatformResourcesTask.onlyIf(java.lang.String, org.gradle.api.specs.Spec)'
	at dev.icerock.gradle.extra.MokoResourcesGenTaskKt$registerGenerateTask$taskProvider$1.invoke(mokoResourcesGenTask.kt:110)
	at dev.icerock.gradle.extra.MokoResourcesGenTaskKt$registerGenerateTask$taskProvider$1.invoke(mokoResourcesGenTask.kt:78)
	at dev.icerock.gradle.extra.MokoResourcesGenTaskKt.registerGenerateTask$lambda$0(mokoResourcesGenTask.kt:81)
	at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:44)
	at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:44)`
	
What I did:

- moved resources from src\commonMain\resources\MR to src\commonMain\moko-resources
- changed dependency in build.gradle:
`buildscript {
    dependencies {
        [...]
        classpath: "dev.icerock.moko:resources-generator:0.24.0"
    }    
}`
- Changed multiplatformResources in build.gradle.kts:
`multiplatformResources {
    resourcesPackage.set("com.example.projectname")
    iosBaseLocalizationRegion.set("en")
}`
- Updated the dependency version in build.gradle.kts to 0.24.0
- Kotlin 1.9.20
- Android Studio Iguana | 2023.2.1 Patch 1

Somehow there seems to be an issue with caching, because the IDE says that the resourcesPackage property of multiplatformResources in the build.gradle.kts file is an unresolved reference. If I go to the declaration of the multiplatformResources it opens the old 0.23.0.jar
I cleared various cache folders, including the ~/.gradle/ folder, which contains the ~/.gradle/caches/jars-9/..../resources-generator-0.24.0.jar and 0.23.0.jar, but to no avail.

Any ideas, what might be wrong?

If I change back the resources-generator class path dependency to 0.23.0 it compiles, but of course all resources are empty, because they are located in the new directory moko-resources, where the old generator does not search for them.

**Comment by Alex009** [comment]
Hi. What gradle version used?

**Comment by bravo-delta** [comment]
Hello, I'm running gradle 7.5.

Thanks for the pointer, that seems to be the issue.
It looks like the update to Kotlin 1.9.20 [requires gradle 8.5](https://docs.gradle.org/current/userguide/compatibility.html)

The error went away.
I was only looking at the minimum requirement of Gradle 7.5 that is mentioned in the moko-resources description and did not look at the one that Kotlin requires.


There is a new error now, but that seems to be related to a resource, where the previous moko version was less restrictive maybe.

`org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':shared:generateMRandroidMain'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:148)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:146)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:134)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at dev.icerock.gradle.utils.StringExtKt.convertXmlStringToAndroidLocalization(StringExt.kt:66)
	at dev.icerock.gradle.generator.resources.string.AndroidStringResourceGenerator.generateLanguageFile(AndroidStringResourceGenerator.kt:73)
	at dev.icerock.gradle.generator.resources.string.AndroidStringResourceGenerator.generateResourceFiles(AndroidStringResourceGenerator.kt:54)
	at dev.icerock.gradle.generator.ResourceTypeGenerator.generateFiles(ResourceTypeGenerator.kt:179)
	at dev.icerock.gradle.generator.ResourcesGenerator.generateResources(ResourcesGenerator.kt:197)
	at dev.icerock.gradle.tasks.GenerateMultiplatformResourcesTask.generate(GenerateMultiplatformResourcesTask.kt:166)`
	
I'll see if I can figure that out.

**Comment by bravo-delta** [comment]
Most likely it has to do with this change: [StringExt.kt String.convertXmlStringToAndroidLocalization](https://github.com/icerockdev/moko-resources/commit/196a1d81dc3ae5df70f6b26f9e961ae96c647e4b#diff-babb32eed09aa9bf3872adb44d2609a8919934e7ccb542d6148297eab9dd57e3)

I have various strings that are empty, e.g. like:
`<string name="empty_string"></string>`

If these are getting into that method the call to this[0] fails.

**Comment by AntsyLich** [comment]
Another example with plurals
```xml
<plurals name="missing_chapters">
    <item quantity="one"/>
    <item quantity="other">Жоқ %1$s тараулары</item>
</plurals>
```

**Comment by Alex009** [comment]
will be released in 0.24.1
